docs/doc-comments-guide.md

```diff
@@ -0,0 +1,90 @@
+     * 
+     * If `false`, the request returns an error if any wildcard expression, index alias, or _all value targets only
+     * missing or closed indices. This behavior applies even if the request targets other open indices. For example,
+     * a request targeting `foo*,bar*` returns an error if an index starts with `foo` but no index starts with `bar`.
+     * 
+     * @server_default true
+     */
+    allow_no_indices?: boolean
+    
+    expand_wildcards?: ExpandWildcards
+    
+    /**
+     * If `true`, missing or closed indices are not included in the response.
+     * @server_default false
+     */
+    ignore_unavailable?: boolean
+    
+    search_type?: string
+  }
+  body: {
+    /**
+     * A set of typical search requests, together with their provided ratings.
+     */
+    requests: RankEvalRequestItem[]
+    
+    /**
+     * Definition of the evaluation metric to calculate.
+     * 
+     * @doc_url https://www.elastic.co/guide/en/elasticsearch/reference/current/search-rank-eval.html#_available_evaluation_metrics
+     */
+    metric?: RankEvalMetric
+  }
+}
+```
+
+([original source code](https://github.com/elastic/elasticsearch-specification/blob/main/specification/_global/rank_eval/RankEvalRequest.ts))
+
+
+## Markup language
+
+The specification is the starting point for a number of generators that produce a large range of artifacts, starting with source code in 10 different programming languages. It's essential for the API documentation to be part of these generated artifacts, e.g. to provide autocompletion help in IDEs when using a client library.
+
+Each of these targets has a different way to format inline documentation: Java code uses HTML, Python uses reStructuredText, Rust uses Markdown, etc. To enable rich documentation to be output, the specification doc comments must not be just plain text, but use a well-defined markup language so that generators can reliably transcode them to their target representation.
+
+**We have settled on using [GitHub Flavored Markdown](https://github.github.com/gfm/) (GFM) for doc comments.** It is based on the well-defined [CommonMark](https://commonmark.org/) specification with a few additions, most notably tables.
+
+GFM also has implementations in most languages, meaning that code generators will be able to easily parse the doc comments and transcode them.
+
+## Structuring a doc-comment
+
+### Terseness
+
+**Doc comments are reference material**: they should be as succinct as possible while capturing all the necessary information to use the elements they're documenting. Remember that they will often show up in small IDE autocompletion popups!
+
+In particular, doc comments are not the right place for tutorials or examples, which should be in dedicated documentation pages. These pages can of course be linked from the doc comments.
+
+API endpoints will also have a `@doc_url` JSDoc tag that links to that API's detailed documentation page.
+
+### Multi-paragraph doc comments
```

Comment:
Can we merge the content with [this section](https://github.com/elastic/elasticsearch-specification/blob/main/docs/modeling-guide.md#description)?

Reply:
Added a [commit](https://github.com/elastic/elasticsearch-specification/pull/638/commits/8483d278157c687fc00f1224237f5d83bd9cbd72) that that adds a link to this document in that section.